coq no longer builds on old gcc

<!-- Thank you for reporting a bug to Coq! -->

#### Description of the problem

It seems that #14519 broke the build of the VM on Ubuntu 14.04 with gcc ... 4.8.4?  (I can't quite tell if it's 4.8 or 4.9).  The full log is [here](https://launchpadlibrarian.net/634145272/buildlog_ubuntu-trusty-amd64.coq_8.master~git~202211171519+23634-0~daily385-e043c3e206~ubuntu14.04.1_BUILDING.txt.gz)

```
$ (cd _build/default/kernel/byterun && /usr/bin/x86_64-linux-gnu-gcc -fPIC -Wall -Wno-unused -g -O2 -msse2 -mfpmath=sse -g -I /usr/lib/ocaml -o coq_interp.o -c coq_interp.c)
In file included from coq_interp.c:51:0:
coq_uint63_native.h: In function ‘uint63_div21_aux’:
coq_uint63_native.h:145:3: error: ‘for’ loop initial declarations are only allowed in C99 mode
   for (int i = 0; i < 63; ++i) {
   ^
coq_uint63_native.h:145:3: note: use option -std=c99 or -std=gnu99 to compile your code
```

cc @ejgallego as author of #14519

#### Coq Version

e043c3e206

--- a/dune-project
+++ b/dune-project
@@ -5,10 +5,6 @@
 (formatting
  (enabled_for ocaml))
 
-; See discussion in https://github.com/coq/coq/pull/14519
-; we handle our CC flags at the low-level ourselves
-(use_standard_c_and_cxx_flags true)
-
 ; After Dune 2.8 we can use this, but let's wait for the build consolidation PR
 ; (generate_opam_files true)
 

--- a/kernel/byterun/dune
+++ b/kernel/byterun/dune
@@ -5,7 +5,7 @@
  (foreign_stubs
   (language c)
   (names coq_fix_code coq_float64 coq_memory coq_values coq_interp)
-  (flags -fPIC (:include %{project_root}/config/dune.c_flags))))
+  (flags :standard (:include %{project_root}/config/dune.c_flags))))
 
 (rule
  (targets coq_instruct.h)

